californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -1041,27 +1041,32 @@ private Block2BlockwiseStatus getBlock2Status(final KeyUri key) {
 	}
 
 	private Block1BlockwiseStatus clearBlock1Status(final KeyUri key) {
+		int size;
+		Block1BlockwiseStatus removedTracker;
 		synchronized (block1Transfers) {
-			Block1BlockwiseStatus removedTracker = block1Transfers.remove(key);
-			if (removedTracker != null) {
-				LOGGER.debug("removing block1 tracker [{}], block1 transfers still in progress: {}",
-						new Object[] { key, block1Transfers.size() });
-				removedTracker.setComplete(true);
-			}
-			return removedTracker;
+			removedTracker = block1Transfers.remove(key);
+			size = block1Transfers.size();
+		}
+		if (removedTracker != null) {
+			LOGGER.debug("removing block1 tracker [{}], block1 transfers still in progress: {}", key, size);
+			removedTracker.setComplete(true);
 		}
+		return removedTracker;
 	}
 
 	private Block2BlockwiseStatus clearBlock2Status(final KeyUri key) {
+		int size;
+		Block2BlockwiseStatus removedTracker;
 		synchronized (block2Transfers) {
-			Block2BlockwiseStatus removedTracker = block2Transfers.remove(key);
-			if (removedTracker != null) {
-				LOGGER.debug("removing block2 tracker [{}], block2 transfers still in progress: {}",
-						new Object[] { key, block2Transfers.size() });
-				removedTracker.setComplete(true);
-			}
-			return removedTracker;
+			removedTracker = block2Transfers.remove(key);
+			size = block2Transfers.size();
+		}
+		if (removedTracker != null) {
+			LOGGER.debug("removing block2 tracker [{}], block2 transfers still in progress: {}",
+					new Object[] { key, size });
```

Comment:
Can we remove the `new Object[]`? It seems not to be any longer required.

Reply:
I will remove the `Object[]`.

By the way do you understand the log message ?
First part ok : _removing block2 tracker [{}]_
Second part : _block2 transfers still in progress: {size}_ sounds a bit strange to me ...

Reply:
replace "still" with "left"?